### PR TITLE
#1806 Provide configurable search count metrics to be exposed via Prometheus

### PR DESCRIPTION
**Pull request by thjaeckle**
- make a common metric prefix configurable for all gathered Ditto custom metrics

- [x] Add and apply configuration for "searches" which should provide "count" metrics as Gauges in search
- [x] Include a "default" search/count, counting all things
- [x] Add documentation and example
- [x] provide configuration of those metrics via Helm chart values

Resolves: #1806

### CODE REVIEW

**Comment by thjaeckle** [comment]
@alstanchev @kalinkostashki I would appreciate a review if you find the time :)

File: documentation/src/main/resources/pages/ditto/installation-operating.md [hunk 1]
```diff
@@ -423,6 +423,61 @@ Have a look at the
 [example Grafana dashboards](https://github.com/eclipse-ditto/ditto/tree/master/deployment/operations/grafana-dashboards)
 and build and share new ones back to the Ditto community.
 
+### Operator defined custom metrics
+
+Staring with Ditto 3.5.0, it is possible to configure "custom metrics" which are gathered by counting things matching
```
[the remark this answers is not in the document]
Comment (thjaeckle) [reply documentation/src/main/resources/pages/ditto/installation-operating.md]: Thx, fixed 👍

File: thingsearch/api/src/main/java/org/eclipse/ditto/thingsearch/api/commands/sudo/SudoCountThings.java [hunk 2]
```diff
@@ -74,7 +95,20 @@ private SudoCountThings(final DittoHeaders dittoHeaders, @Nullable final String
...
     public static SudoCountThings of(@Nullable final String filter, final DittoHeaders dittoHeaders) {
-        return new SudoCountThings(dittoHeaders, filter);
+        return new SudoCountThings(dittoHeaders, filter, null);
+    }
+
+    /**
+     * Returns a new instance of {@code SudoCountThings}.
+     *
+     * @param filter the optional filter string
+     * @param dittoHeaders the headers of the command.
```
Comment (alstanchev) [inline thingsearch/api/src/main/java/org/eclipse/ditto/thingsearch/api/commands/sudo/SudoCountThings.java:105]: Missing doc for namespaces

**Review by alstanchev — APPROVED** [review]
Other then the minor java doc for parameter 
looks realy good 👍

File: thingsearch/api/src/main/java/org/eclipse/ditto/thingsearch/api/commands/sudo/SudoCountThings.java (hunk 2 above)
Comment (thjaeckle) [reply thingsearch/api/src/main/java/org/eclipse/ditto/thingsearch/api/commands/sudo/SudoCountThings.java:106]: Yeah, good catch - fixed :)

**Pull request merged**